test_lib.py

```diff
@@ -337,44 +337,129 @@ def test_add_service_urls_using_metamodel(self):
-       
-        #case 2: checking for package_dict{}
-        service_urls_map = { 'https://vcip/rest/vmware/com/package/mock' : 'com.vmware.package.mock'}
+        self.assertEqual({}, package_dict_actual)
+
+        # case 1.2: -dsr on
+        package_dict_api_actual, \
+        package_dict_actual, _, _, = dict_processing.add_service_urls_using_metamodel(service_urls_map,
+                                                                                      service_dict,
+                                                                                      rest_navigation_handler,
+                                                                                      True)
+        self.assertEqual(package_dict_api_expected, package_dict_api_actual)
+        self.assertEqual({}, package_dict_actual)
+
+        # case 2: /api operation and /rest equivalent (RestNavigation)
+        # Rest navigation returns not None
+        rest_navigation_handler.get_service_operations = mock.MagicMock(return_value={})
+        # case 2.1: -dsr off
+        package_dict_api_actual, \
+        package_dict_actual, _, _, = dict_processing.add_service_urls_using_metamodel(service_urls_map,
+                                                                                      service_dict,
+                                                                                      rest_navigation_handler)
+        package_dict_expected = {'package': ['/com/vmware/package/mock']}
+        self.assertEqual({}, package_dict_api_actual)
+        self.assertEqual(package_dict_expected, package_dict_actual)
+
+        # case 2.2 -dsr on
+        package_dict_api_actual, \
+        package_dict_actual, \
+        package_dict_deprecated_actual, _, = dict_processing.add_service_urls_using_metamodel(service_urls_map,
+                                                                                              service_dict,
+                                                                                              rest_navigation_handler,
+                                                                                              True)
+        package_dict_expected = {'package': ['/com/vmware/package/mock']}
+        package_dict_api_expected = {'package': ['/package/mock']}
+        self.assertEqual(package_dict_api_expected, package_dict_api_actual)
+        self.assertEqual(package_dict_expected, package_dict_deprecated_actual)
+        self.assertEqual({}, package_dict_actual)
+
+        # case 3: /rest operation only
+        service_urls_map = {'https://vcip/rest/vmware/com/package/mock': 'com.vmware.package.mock'}
         element_value_mock.string_value = 'mock_string_value'
-        operation_info_mock.metadata = { 
-                            'mock_element_key' : element_info_mock
-                            }
+        operation_info_mock.metadata = {
+            'mock_element_key': element_info_mock
+        }
+        # case 3.1: -dsr off
```

Comment:
please use the full name of the switch, saving several characters is not worth it the troubles you are going to cause to the next reader of the code (who doesn't know the switches by heart, like you right now).

There are some more occurrences of this issue around,,,

Reply:
Added the full name.